staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -339,10 +349,24 @@ type fsStatusUpdate struct {
 
 // digestConfigObjects is given all the API objects that configure
 // cfgCtlr and writes its consequent new configState.
-func (cfgCtlr *configController) digestConfigObjects(newPLs []*flowcontrol.PriorityLevelConfiguration, newFSs []*flowcontrol.FlowSchema) error {
+func (cfgCtlr *configController) digestConfigObjects(newPLs []*flowcontrol.PriorityLevelConfiguration, newFSs []*flowcontrol.FlowSchema) (time.Duration, error) {
 	fsStatusUpdates := cfgCtlr.lockAndDigestConfigObjects(newPLs, newFSs)
 	var errs []error
+	currResult := updateResult{
+		timeUpdated:  time.Now(),
+		updatedItems: sets.String{},
+	}
+	suggestedDelay := 0 * time.Minute
 	for _, fsu := range fsStatusUpdates {
+		// if we should skip this name, indicate we will need a delay, but continue with other entries
+		if cfgCtlr.shouldDelayUpdate(fsu.flowSchema.Name) {
+			suggestedDelay = 1 * time.Minute // our memory is only one minute long.
```

Comment:
In a mixed-version scenario, where we have two competing gangs of apiservers, they are likely to arrange themselves into two thundering herds.  Well, "thundering" is a bit strong, but my point here is about the synchronization.  If there were randomization in the suggested delay, that would defeat the synchronization.